tests/10-issue.js

```diff
@@ -64,206 +69,248 @@ describe('Issuers - BitstringStatusList',
-                      equal(issuedVc.credentialStatus.statusSize,
-                        'Expected statusMessage lenght to be equal to ' +
-                        'statusSize.');
-                  });
-                }
-                if(Number.isInteger(issuedVc.credentialStatus.statusSize) &&
-                issuedVc.credentialStatus.statusSize.gt(1)) {
-                  it('statusMessage MAY be present if statusSize is 1, ' +
-                  'and MUST be present if statusSize is greater than 1.',
-                  async function() {
-                    this.test.link = 'https://www.w3.org/TR/vc-bitstring-status-list/#:~:text=statusMessage%20MAY%20be%20present%20if%20statusSize%20is%201%2C%20and%20MUST%20be%20present%20if%20statusSize%20is%20greater%20than%201';
-                    issuedVc.credentialStatus.should.have.own.property(
-                      'statusMessage');
-                  });
-                }
-                it('If the statusMessage array is present, each ' +
-                  'element MUST contain the two properties "status" ' +
-                  'and "message".',
-                async function() {
-                  this.test.link = 'https://www.w3.org/TR/vc-bitstring-status-list/#:~:text=If%20the%20statusMessage%20array%20is%20present%2C%20each%20element%20MUST%20contain%20the%20two%20properties%20described%20below%2C%20and%20MAY%20contain%20additional%20properties.';
-                  // TODO this needs more testing:
-                  // a string representing the hexadecimal
-                  // value of the status prefixed with 0x
-                  const statusMessage = issuedVc.credentialStatus.statusMessage;
+                // TODO test for integer
+                assert(Number.isInteger(
+                  statusEntries[statusEntry].statusSize),
+                'Expected statusSize to be an integer.');
+                statusEntries[statusEntry].statusSize.should.be.gt(0,
+                  'Expected statusSize to be greater than zero.');
+              } else {
+                this.skip();
+              }
+            }
+          });
+          it('If statusSize is provided and is greater than 1, ' +
+            'then the property credentialStatus.statusMessage ' +
+            'MUST be present.',
+          async function() {
+            this.test.link = 'https://www.w3.org/TR/vc-bitstring-status-list/#:~:text=If%20statusSize%20is%20provided%20and%20is%20greater%20than%201%2C%20then%20the%20property%20credentialStatus.statusMessage%20MUST%20be%20present';
+            for(statusEntry in statusEntries) {
+              if('statusSize' in statusEntries[statusEntry] &&
+                Number.isInteger(statusEntries[statusEntry].statusSize) &&
+                statusEntries[statusEntry].statusSize.gt(1)) {
+                statusEntries[statusEntry].should.have.own.property(
+                  'statusMessage');
+              } else {
+                this.skip();
```

Comment:
so if a single stratusEntry lacks a `statusSize` we skip the entire test? I would use `continue` and check the next statusEntry.

Reply:
I added a check if it's not on the last element of the array it will continue. I want to avoid reporting feature support for unsupported features. Simply having a continue would make it seem like the `statusMessage` was supported even if it's not